### DISCUSSION TITLE
Recommendations for E2E testing addins

### DISCUSSION THREAD

**Discussion by schmkr** [OfficeDev/office-js#4201] [Q&A]
Hello, 

For our add-in in Outlook, we have written End-to-End-tests using [Playwright](https://playwright.dev/). This works just fine, even with logging in and MFA. 

One thing we have run into though, is it seems Playwright needs to be running in _headed_ mode and cannot run _headless_. Whenever it runs headless, the user ends up in a different (more basic/lightweight) version of Outlook online:

![Screenshot 2024-03-01 at 14 48 37](https://github.com/OfficeDev/office-js/assets/612280/880ff227-960d-4553-a8ef-fb5210202f73)

I am aware some websites implement anti-headless-browsers measures and that there are ways to circumvent those measures. Unfortunately, without success, up to now. 

So I was curious, to hear from the community or Microsoft, is there a way to ensure even an headless browser ends up on the "modern" or "regular" online Outlook, even when using a headless browser? 

This will allow us to ensure our add-in for Outlook online customers functions as expected and does not break user's experience.

**Comment by merkelfleet** [comment]
Hey. I don't have any answers to your questions, but I have a few questions of my own.
I also would like to add E2E tests for our Excel add-in. There is no documentation about this topic on the official pages.

How did you created your Playwright environment and the tests? And how do you use your add-in in the tests? Is there a guide I can follow?

**Reply by YossiCohen** [reply]
Any conclusions related to that?

**Reply by kamalprakhar** [reply]
hey @merkelfleet 
did you find any guide regarding integration testing?

**Reply by merkelfleet** [reply]
Sorry for my late answer. No, the project is no longer my focus.

**Comment by kamalprakhar** [comment]
hey @schmkr 
can you please help with any guide/docs, we're planning to have integration testing for our excel-add-in.

**Reply by schmkr** [reply]
hi @kamalprakhar, I am no longer with the company that needed this.
